tmk_core/protocol/chibios/chibios.c

```diff
@@ -62,14 +62,47 @@
  */
 
 /* declarations */
-uint8_t keyboard_leds(void);
-void    send_keyboard(report_keyboard_t *report);
-void    send_nkro(report_nkro_t *report);
-void    send_mouse(report_mouse_t *report);
-void    send_extra(report_extra_t *report);
+extern uint8_t keyboard_leds(void);
+extern void    send_keyboard(report_keyboard_t *report);
+extern void    send_nkro(report_nkro_t *report);
+extern void    send_mouse(report_mouse_t *report);
+extern void    send_extra(report_extra_t *report);
+#ifdef JOYSTICK_ENABLE
+extern void send_joystick(report_joystick_t *report);
+#endif // JOYSTICK_ENABLE
+#ifdef DIGITIZER_ENABLE
+extern void send_digitizer(report_digitizer_t *report);
+#endif // DIGITIZER_ENABLE
+#ifdef PROGRAMMABLE_BUTTON_ENABLE
+extern void send_programmable_button(report_programmable_button_t *report);
+#endif // PROGRAMMABLE_BUTTON_ENABLE
 
 /* host struct */
-host_driver_t chibios_driver = {keyboard_leds, send_keyboard, send_nkro, send_mouse, send_extra};
+host_driver_t chibios_driver = {
+    .has_init_executed = false,
+    .init              = NULL,
+    .connect           = NULL,
+    .disconnect        = NULL,
+    .is_connected      = NULL,
```

Comment:
Should this use `usb_connected_state()`?

Reply:
Probably -- I wouldn't read too much into the current implementation right now; it's raised as a draft for visibility and discussion purposes. Nowhere near ready for merge.